engine/src/gnuhash.h

```diff
@@ -2,63 +2,22 @@
 #define _GNUHASH_H_
 #include "config.h"
 #include <unordered_map>
+#include <functional>
 
 #define vsUMap     std::unordered_map
 #define vsHashComp std::hash_compare
 #define vsHash     std::hash
-class Unit;
 
 namespace std
 {
-template < >
-class hash< void* >
-{
-    hash< size_t >a;
-public:
-    size_t operator()( const void *key ) const
-    {
-        return a( (size_t) key );
-    }
-};
-template < >
-class hash< const void* >
-{
-    hash< size_t >a;
-public:
-    size_t operator()( const void*const &key ) const
-    {
-        return a( (size_t) key );
-    }
-};
-template < >
-class hash< const Unit* >
-{
-    hash< size_t >a;
-public:
-    size_t operator()( const Unit*const &key ) const
-    {
-        return a( (size_t) key>>4 );
-    }
-};
-template < >
-class hash< std::pair< Unit*, Unit* > >
-{
-    hash< size_t >a;
-public:
-    size_t operator()( const std::pair< Unit*, Unit* > &key ) const
-    {
-        return (size_t) (size_t) ( a( (int) ( ( (size_t) key.first )>>4 ) )
-                                  ^a( (int) ( ( (size_t) key.second )>>4 ) ) );
-    }
```

Comment:
Since this is operating on the Unit this should probably be kept, though perhaps defined in Unit.